fedimint-server/src/consensus/mod.rs

```diff
@@ -143,10 +143,14 @@ impl FedimintConsensus {
             };
             info!(module_instance_id = *module_id, kind = %kind, "Init module");
 
+            let isolated_db = db.new_isolated(*module_id);
+            fedimint_api::db::apply_migrations(&isolated_db, init.get_database_migration_data())
```

Comment:
We could probably move `fedimint_api::db:apply_migrations` to `fedimint-server` now.

Reply:
Moved to `fedimint-server`

Reply:
I moved `apply_migrations` back to `fedimint_api`. I moved it back so that the unit test could be in `fedimint_api`. It also occurred to me that the client will want to have the ability to call `apply_migrations` too, so I think keeping it in `fedimint_api` is ok.